### PR TITLE
chore: bump reth, starknet and starknet api

### PR DESCRIPTION
**Pull request by Eikix**
Time spent on this PR:

Resolves: #654 

# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

# What is the new behavior?

# Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: crates/core/src/client/errors.rs [hunk 1]
```diff
@@ -46,10 +46,10 @@ pub enum ConfigError {
 
 /// Error that can accure when interacting with the Kakarot ETH API.
 #[derive(Debug, Error)]
-pub enum EthApiError<E: std::error::Error> {
+pub enum EthApiError {
     /// Request to the Starknet provider failed.
     #[error(transparent)]
-    RequestError(#[from] ProviderError<E>),
```
Comment (greged93) [inline crates/core/src/client/errors.rs:52]: Finally this was removed lol

File: crates/core/src/client/helpers.rs [hunk 2]
```diff
@@ -127,7 +95,7 @@ pub fn raw_kakarot_calldata(kakarot_address: FieldElement, mut calldata: Vec<Fie
 /// Helper function to split a U256 value into two FieldElements.
 pub fn split_u256_into_field_elements(value: U256) -> [FieldElement; 2] {
     let low = value & U256::from(U128::MAX);
-    let high = value >> 128;
+    let high: U256 = value >> 128;
```
Comment (greged93) [inline crates/core/src/client/helpers.rs:98]: did you have to add this?

File: crates/core/src/client/mod.rs [hunk 3]
```diff
@@ -215,7 +210,7 @@ impl<P: Provider + Send + Sync + 'static> KakarotClient<P> {
         let key_low = index & U256::from(u128::MAX);
         let key_low: Felt252Wrapper = key_low.try_into()?;
 
-        let key_high = index >> 128;
+        let key_high: U256 = index >> 128;
```
Comment (greged93) [inline crates/core/src/client/mod.rs:213]: same here, do we need this?

File: crates/core/src/models/transaction/transaction.rs [hunk 4]
```diff
@@ -103,7 +103,7 @@ impl ConvertibleStarknetTransaction for StarknetTransaction {
             },
             _ => return Err(EthApiError::KakarotDataFilteringError("Transaction".into())),
         };
-        let nonce: U256 = nonce.into();
+        let nonce: U64 = u64::try_from(nonce).expect("INVARIANT: nonce should be a u64").into();
```
Comment (greged93) [inline crates/core/src/models/transaction/transaction.rs:106]: This looks very weird, why are we doing this? I don't think the RPC should be doing the job of checking if nonces are u64

File: rust-toolchain [hunk 5]
```diff
@@ -1 +1,2 @@
-nightly-2023-06-18
+[toolchain]
+channel = "1.73.0"
```
Comment (greged93) [inline rust-toolchain:2]: You're missing a new line

**Review by greged93 — COMMENTED** [review]
finally `ProviderError` doesn't take a generic

File: crates/core/src/models/transaction/transaction.rs (hunk 4 above)
Comment (Eikix) [reply crates/core/src/models/transaction/transaction.rs:106]: Eth Api expects a U64, whereas Starknet RPC returns a FieldElement, so we need to make one fit into the other

File: crates/core/src/client/mod.rs (hunk 3 above)
Comment (Eikix) [reply crates/core/src/client/mod.rs:213]: compiler needs help weirdly

File: crates/core/src/models/transaction/transaction.rs (hunk 4 above)
Comment (Eikix) [reply crates/core/src/models/transaction/transaction.rs:106]: Ok will change for "?"

**Review by greged93 — DISMISSED** [review]
lgtm!

**Review by greged93 — APPROVED** [review]
still lgtm

**Pull request merged**